### PR TITLE
#1793 Add AmazonIoTClient and AmazonIotDataClient configurations

### LINKED ISSUE #1793: Task: Add AmazonIoTClient and AmazonIotDataClient configurations to interact with AWS IoT services
**Is your feature request related to a problem? Please describe.**
IoT Hub Portal must be able to interact with AWS IoT services.

**Describe the solution you'd like**

- Install the nuget package [`AWSSDK.IoT`](https://www.nuget.org/packages/AWSSDK.IoT) on backend
- Create a new singleton of [`AmazonIoTClient`](https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/IoT/TIoTClient.html). 
- Install the nuget package [`AWSSDK.IotData`](https://www.nuget.org/packages/AWSSDK.IotData) on backend
- Create a new singleton of [`AmazonIotDataClient`](https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/IotData/TIotDataClient.html). 

Suggestion:

```csharp
// Get AWS infos from configuraiton
var accessKey = "changeme";
var secretAccessKey = "changeme";
var region = "changeme";

_ = services.AddSingleton(new AmazonIoTClient(accessKey, secretAccessKey, RegionEndpoint.GetBySystemName(region)));
_ = services.AddSingleton(new AmazonIotDataClient(accessKey, secretAccessKey, new AmazonIotDataConfig()));
```

### PR DESCRIPTION
**Pull request by delager**
- Add "CloudProvider" in configuration
- Add AmazonIoTClient and AmazonIotDataClient configurations when CloudProvider is AWS

## Description

What's new?

- Fix #1793

## What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Tests
- [ ] Other

### CODE REVIEW

File: src/AzureIoTHub.Portal.Infrastructure/DevelopmentConfigHandler.cs [hunk 1]
```diff
@@ -82,6 +82,8 @@ internal DevelopmentConfigHandler(IConfiguration config)
 
         public override string DbProvider => this.config.GetValue(DbProviderKey, DbProviders.PostgreSQL);
 
+        public override string CloudProvider => this.config[CloudProviderKey];
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal.Infrastructure/DevelopmentConfigHandler.cs:85]: Set by default `Azure`

```suggestion
        public override string CloudProvider => this.config.GetValue(CloudProviderKey, "Azure");
```

File: src/AzureIoTHub.Portal.Infrastructure/ProductionConfigHandler.cs [hunk 2]
```diff
@@ -82,6 +82,8 @@ internal ProductionConfigHandler(IConfiguration config)
         public override string IdeasAuthenticationHeader => this.config.GetValue(IdeasAuthenticationHeaderKey, "Ocp-Apim-Subscription-Key");
         public override string IdeasAuthenticationToken => this.config.GetValue(IdeasAuthenticationTokenKey, string.Empty);
 
+        public override string CloudProvider => this.config[CloudProviderKey];
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal.Infrastructure/ProductionConfigHandler.cs:85]:
```suggestion
        public override string CloudProvider => this.config.GetValue(CloudProviderKey, "Azure");
```

**Review by kbeaugrand — CHANGES_REQUESTED** [review]

File: src/AzureIoTHub.Portal.Infrastructure/DevelopmentConfigHandler.cs (hunk 1 above)
Comment (delager) [reply src/AzureIoTHub.Portal.Infrastructure/DevelopmentConfigHandler.cs:85]: @kbeaugrand It was decided to not set a default value for the CloudProvider but to generate an exception if the configuration was not present (task #1928). 
We still set Azure as default value ?

File: src/AzureIoTHub.Portal.Infrastructure/ProductionConfigHandler.cs (hunk 2 above)
Comment (delager) [reply src/AzureIoTHub.Portal.Infrastructure/ProductionConfigHandler.cs:85]: @kbeaugrand It was decided to not set a default value for the CloudProvider but to generate an exception if the configuration was not present (task #1928). 
We still set Azure as default value ?

Comment (kbeaugrand) [reply src/AzureIoTHub.Portal.Infrastructure/ProductionConfigHandler.cs:85]: @delager you're right. We should instead provide the value manually in the e2e-docker-compose.yml file to make the tests working.

**Review by kbeaugrand — APPROVED** [review]

**Pull request merged**